homeassistant/components/incomfort/config_flow.py

```diff
@@ -0,0 +1,51 @@
+"""Config flow support for Intergas InComfort integration."""
+
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_USERNAME
+from homeassistant.helpers.selector import (
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import DOMAIN
+from .models import async_connect_gateway
+
+TITLE = "Intergas InComfort/Intouch Lan2RF gateway"
+
+
+class InComfortConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Config flow to set up an Intergas InComfort boyler and thermostats."""
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors: dict[str, str] = {}
+        data_schema = {
+            vol.Required(CONF_HOST): TextSelector(
+                TextSelectorConfig(type=TextSelectorType.TEXT)
+            ),
+            vol.Optional(CONF_USERNAME): TextSelector(
+                TextSelectorConfig(type=TextSelectorType.TEXT, autocomplete="admin")
+            ),
+            vol.Optional(CONF_PASSWORD): TextSelector(
+                TextSelectorConfig(type=TextSelectorType.PASSWORD)
+            ),
+        }
+        if user_input is not None:
+            self._async_abort_entries_match({CONF_HOST: user_input[CONF_HOST]})
+            if await async_connect_gateway(self.hass, user_input, errors):
+                return self.async_create_entry(title=TITLE, data=user_input)
+
+        return self.async_show_form(
+            step_id="user", data_schema=vol.Schema(data_schema), errors=errors
+        )
+
+    async def async_step_import(self, import_data: dict[str, Any]) -> ConfigFlowResult:
+        """Import `incomfort` config entry from configuration.yaml."""
+        return self.async_create_entry(title=TITLE, data=import_data)
```

Comment:
The import step can currently return a form when the config is invalid. We should avoid that and abort in that case.

Reply:
The YAML config is validated already and only imported if there was no config entry defined yet. How can the config be invalid?

Reply:
Password incorrect, device doesn't exist anymore, hostname not found.

Whatever issue that would break current usage already, but for people who don't notice this. According to ADR-0021 we should give them a good issue saying what went wrong and that they should fix it

Reply:
Added error handling and tests on import code